docs/architecture/maui/accessing-remote-data.md

```diff
@@ -1,9 +1,380 @@
+
+The following code example shows the `GetAsync` method in the `RequestProvider` class:
+
+```csharp
+public async Task<TResult> GetAsync<TResult>(string uri, string token = "")
+{
+    HttpClient httpClient = CreateHttpClient(token);
+    HttpResponseMessage response = await httpClient.GetAsync(uri);
+
+    await HandleResponse(response);
+    string serialized = await response.Content.ReadAsStringAsync();
+
+    TResult result = await Task.Run(() => 
+        JsonConvert.DeserializeObject<TResult>(serialized, _serializerSettings));
+
+    return result;
+}
+```
+
+This method calls the `CreateHttpClient` method, which returns an instance of the `HttpClient` class with the appropriate headers set. It then submits an asynchronous `GET` request to the resource identified by the URI, with the response being stored in the `HttpResponseMessage` instance. The `HandleResponse` method is then invoked, which throws an exception if the response doesn't include a success HTTP status code. Then the response is read as a string, converted from JSON to a `CatalogRoot` object, and returned to the `CatalogService`.
+
+The `CreateHttpClient` method is shown in the following code example:
+
+```csharp
+private HttpClient _httpClient;
+
+private HttpClient CreateHttpClient(string token = "")
+{
+    if(_httpClient == null)
+    {
+        _httpClient = new HttpClient();
+        _httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+    }
+
+    if (!string.IsNullOrEmpty(token))
+    {
+        _httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+    }
+
+    return _httpClient;
+}
+```
+
+This method uses creates a new instance or retrieves a cached instance of the `HttpClient` class, and sets the Accept header of any requests made by the `HttpClient` instance to `application/json`, which indicates that it expects the content of any response to be formatted using JSON. Then, if an access token was passed as an argument to the `CreateHttpClient` method, it's added to the `Authorization` header of any requests made by the `HttpClient` instance, prefixed with the string `Bearer`. For more information about authorization, see [Authorization](authentication-and-authorization.md).
+
+> [!TIP]
+> It is highly recommended to cache and reuse instances of the `HttpClient` for better application performance.
```

Comment:
Great callout, but why? You should mention the concerns of socket exhaustion and link to our architecture docs on the matter.

Reply:
Good call. Updated the language and added a pointer to the Instancing docs.